homeassistant/components/knx/weather.py

```diff
@@ -0,0 +1,61 @@
+"""Support for KNX/IP weather station."""
+from xknx.devices import Weather as XknxWeather
+
+from . import DATA_KNX
+from ...const import TEMP_CELSIUS
+from ..weather import WeatherEntity
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the scenes for KNX platform."""
+    entities = []
+    for device in hass.data[DATA_KNX].xknx.devices:
+        if isinstance(device, XknxWeather):
+            entities.append(KNXWeather(device))
+    async_add_entities(entities)
+
+
+class KNXWeather(WeatherEntity):
+    """Representation of a KNX weather device."""
+
+    def __init__(self, device: XknxWeather):
+        """Initialize of a KNX sensor."""
+        self.device = device
+
+    @property
+    def temperature(self):
+        """Return current temperature."""
+        return self.device.temperature
+
+    @property
+    def temperature_unit(self):
+        """Return temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def pressure(self):
+        """Return current air pressure."""
+        # KNX returns pA - HA requires hPa
+        return (
+            self.device.air_pressure / 100
+            if self.device.air_pressure is not None
+            else None
+        )
+
+    @property
+    def condition(self):
+        """Return current weather condition."""
+        return self.device.ha_current_state().value
```

Comment:
Is it required to do an additional mapping here? We only use the HA states defined for the `WeatherEntity`.

https://github.com/XKNX/xknx/blob/0.13.0/xknx/devices/weather.py#L24

Reply:
No, we don't have any constants defined. We just have a recommended set of state conditions:
https://developers.home-assistant.io/docs/core/entity/weather#recommended-values-for-state-and-condition

Reply:
Perfect - that's exactly what I used.